app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,64 @@
+class PostsController < ApplicationController
+
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action(:find_post, {only: [:show, :edit, :update, :destroy]})
+  before_action :authorize_post, only: [:edit, :update, :destroy]
+
```

Comment:
You can just write
before_action :find_post, only: [:show, :edit, :update, :destroy]
to make them consistent.